gax-java/gax/src/main/java/com/google/api/gax/rpc/Watchdog.java

```diff
@@ -318,14 +318,26 @@ boolean cancelIfStale() {
         switch (this.state) {
           case IDLE:
             if (!idleTimeout.isZero() && waitTime >= idleTimeout.toMillis()) {
-              myError = new WatchdogTimeoutException("Canceled due to idle connection", false);
+              myError =
+                  new WatchdogTimeoutException(
+                      "Canceled idle connection after "
+                          + waitTime
+                          + " ms (threshold: "
+                          + idleTimeout.toMillis()
```

Comment:
Now that the error message is not a static String anymore, can you please add some test for it? We already have some [test cases](https://github.com/googleapis/gapic-generator-java/blob/59efbbbd485282e37abbc5eed87b72a15a33240e/gax-java/gax/src/test/java/com/google/api/gax/rpc/WatchdogTest.java#L154) that verify the `WatchdogTimeoutException`, you could add assertions for the error message.